AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSAnalyticsTransmissionTarget.m

```diff
@@ -37,10 +56,49 @@ - (MSAnalyticsTransmissionTarget *)transmissionTargetForToken:(NSString *)token
   // Look up for the token in the dictionary, create a new transmission target if doesn't exist.
   MSAnalyticsTransmissionTarget *target = self.childTransmissionTargets[token];
   if (!target) {
-    target = [[MSAnalyticsTransmissionTarget alloc] initWithTransmissionTargetToken:token];
+    target = [[MSAnalyticsTransmissionTarget alloc] initWithTransmissionTargetToken:token parentTarget:self];
     self.childTransmissionTargets[token] = target;
   }
   return target;
 }
 
+- (BOOL)isEnabled {
+
+  // Get isEnabled value from persistence.
+  // No need to cache the value in a property, user settings already have their cache mechanism.
+  NSNumber *isEnabledNumber = [self.storage objectForKey:self.isEnabledKey];
+
+  // Return the persisted value otherwise it's enabled by default.
+  return (isEnabledNumber) ? [isEnabledNumber boolValue] : YES;
+}
+
+- (void)setEnabled:(BOOL)isEnabled {
+  if (self.isEnabled != isEnabled) {
+
+    // Don't enable if the immediate parent is disabled.
+    if (isEnabled && ![self isAncestorEnabled]) {
+      MSLogWarning([MSAnalytics logTag],
+                   @"Can't enable; parent transmission target and/or Analytics service is disabled.");
+      return;
+    }
+
+    // Persist the enabled status.
+    [self.storage setObject:@(isEnabled) forKey:self.isEnabledKey];
+  }
+
+  // Propagate to nested transmission targets. TODO Find a more effective approach.
+  for (NSString *token in self.childTransmissionTargets) {
+    [self.childTransmissionTargets[token] setEnabled:isEnabled];
+  }
+}
+
+/**
+ * Check ancestor enabled state, the ancestor is either the immediate target parent if there is one or Analytics.
+ *
+ * @return YES if the immediate ancestor is enabled.
+ */
+- (BOOL)isAncestorEnabled {
```

Comment:
Is the method somewhat we've agreed with other platforms? It only checks it's immediate ancestor, it would be better to say parent I guess.
The explanation for the method isn't quite understandable. Can you revise this one?

Reply:
Yeah, we should change the name of this method.  I thought we needed to mirror the behavior in android but it's not necessary since we copy the state of the parent when creating the transmission target.